mkosi/__init__.py

```diff
@@ -3717,13 +3722,19 @@ def calculate_bmap(args: CommandLineArguments, raw: Optional[BinaryIO]) -> Optio
 
 
 def save_cache(args: CommandLineArguments, root: str, raw: Optional[str], cache_path: Optional[str]) -> None:
-    if cache_path is None or raw is None:
+    if cache_path is None:
+        return
+
+    # bail out if cache already exists
+    if os.path.exists(cache_path):
```

Comment:
```suggestion
    if cache_path is None or os.path.exists(cache_path):
        return
```
I think this is clearer than two separate conditionals.